### DISCUSSION TITLE
Possible use of native and ShiftResgister leds toghether in custom board

### DISCUSSION THREAD

**Discussion by azgael** [shanteacontrols/OpenDeck#200] [Q&A] [answered]
Dear Paradajz
I plan to use a Arduino 33 custom setup as base micro controller for a midi controller having:
18 rotary encoder
8 buttons
64 leds + 2 leds internal. 

The intent is to use for 16 encoders using 4x 74HC165 plus 12 GPIO available in the board.
Similar approach for the leds using 64 leds via a max7219 and additional 2 from native GPIO.

My question is is it possible to combine use of "native" GPIO and the shift registers for buttons?
Same to the LED is it possible to combine the max7219 and additional "native GPIO?

I have tried to compile using board customization but seams that as soon as I use shift registers or the max the GPIO are not innumerate by the online configurator any more. 
as example:
"
buttons:
    type: "shiftRegister"
    shiftRegisters: 2
    pins:
      data:
        port: "D"
        index: 3
      clock:
        port: "D"
        index: 4
      latch:
        port: "D"
        index: 5

    type: "native"
     extPullups: true
     pins:
     -
     port: "B"
       index: 15
     -
     port: "B"
       index: 14

leds:
   external:
     type: "native"
     invert: false
     pins:
     -
       port: "B"
       index: 15
     -
       port: "B"
       index: 14

      type: "max7219"
      invert: false
      pins:
        data:
          port: "1"
          index: 8
        clock:
          port: "0"
          index: 4
        latch:
          port: "1"
          index: 11
"

**Comment by paradajz** [comment, marked as answer]
No, combining different drivers is not possible and will never be possible.
